R/summary_funs.R

```diff
@@ -259,141 +233,178 @@ weighted_summary_means <- function(y_wobs_test, family, wdraws, mu, dis, cl_ref,
+                                   y_idx = loo_inds,
+                                   w = wobs)
+      value <- srs_diffe$y_hat / n_full
+      # combine estimates of var(y_hat) and var(y)
+      value_se <- sqrt(srs_diffe$v_y_hat + srs_diffe$hat_v_y) / n_full
+    } else {
+      # full LOO estimator
+      value <- mean(wobs * (mu - y)^2)
+      value_se <- .weighted_sd((mu - y)^2, wobs) / sqrt(n_full)
+    }
+    # store for later calculations
+    mse_e <- value
+    if (!is.null(summaries_baseline)) {
+      # delta=TRUE, variance of difference of two normally distributed
+      mse_b <- mean(wobs * (mu_baseline - y)^2)
+      var_mse_b <- .weighted_sd((mu_baseline - y)^2, wobs)^2 / n_full
+      if (n_loo < n_full) {
+        mse_e_fast <- mean(wobs * (summaries_fast$mu - y)^2)
+        srs_diffe <-
+          .srs_diff_est_w(y_approx = ((summaries_fast$mu - y)^2 - mse_e_fast) *
+                            ((mu_baseline - y)^2 - mse_b),
+                          y = (((mu - y)^2 - mse_e) *
+                                 ((mu_baseline - y)^2 - mse_b))[loo_inds],
+                          y_idx = loo_inds,
+                          w = wobs)
+        cov_mse_e_b <- srs_diffe$y_hat / n_full^2
       } else {
-        value <- mean(wcv * (mu - y)^2, na.rm = TRUE)
-        value.se <- weighted.sd((mu - y)^2, wcv, na.rm = TRUE) /
-          sqrt(n_notna)
+        cov_mse_e_b <- mean(wobs * ((mu - y)^2 - mse_e) *
+                              ((mu_baseline - y)^2 - mse_b)) / n_full
       }
+      value_se <- sqrt(value_se^2 - 2 * cov_mse_e_b + var_mse_b)
+    }
+    if (stat == "mse") {
+      value <- mse_e - ifelse(is.null(summaries_baseline), 0, mse_b)
     } else if (stat == "rmse") {
-      if (!is.null(mu.bs)) {
-        value <- sqrt(mean(wcv * (mu - y)^2, na.rm = TRUE)) -
-          sqrt(mean(wcv * (mu.bs - y)^2, na.rm = TRUE))
-        diffvalue.bootstrap <- bootstrap(
-          cbind((mu - y)^2, (mu.bs - y)^2),
-          function(resid2) {
-            sqrt(mean(wcv * resid2[, 1], na.rm = TRUE)) -
-              sqrt(mean(wcv * resid2[, 2], na.rm = TRUE))
-          },
-          ...
-        )
-        value.se <- sd(diffvalue.bootstrap)
-        lq_uq <- quantile(diffvalue.bootstrap,
-                          probs = c(alpha_half, one_minus_alpha_half),
-                          names = FALSE, na.rm = TRUE)
+      # simple transformation of mse
+      value <- sqrt(mse_e) - ifelse(is.null(summaries_baseline), 0, sqrt(mse_b))
+      # the first-order Taylor approximation of the variance
+      value_se <- sqrt(value_se^2 / mse_e / 4)
+    } else if (stat == "R2") {
+      # simple transformation of mse
+      mse_y <- mean(wobs * (mean(y) - y)^2)
+      value <- 1 - mse_e / mse_y - ifelse(is.null(summaries_baseline), 0, 1 - mse_b / mse_y)
+      # the first-order Taylor approximation of the variance
+      var_mse_y <- .weighted_sd((mean(y) - y)^2, wobs)^2 / n_full
```

Comment:
For `mean(y)`, don't we need to take `wobs` into account? I think this is similar to line `var_mse_b <- .weighted_sd((mu_baseline - y)^2, wobs)^2 / n_full` where the parameter estimates from which `mu_baseline` is computed also take `wobs` into account.

Reply:
This also concerns several other occurrences of `mean(y)` here in `get_stat()`.